tsdb/wal.go

```diff
@@ -102,6 +104,7 @@ type WAL interface {
 type WALReader interface {
 	Read(
 		seriesf func([]record.RefSeries),
+		metadataf func([]record.RefMetadata),
```

Comment:
Does checkpointing need updating for the new data?

Reply:
Ah yes, good catch. Updated (still pending tests though).